skfda/ml/regression/linear_model.py

```diff
@@ -0,0 +1,99 @@
+from sklearn.metrics import mean_squared_error
+from sklearn.base import BaseEstimator, RegressorMixin
+from skfda.representation.basis import *
+
+import numpy as np
+
+
+class LinearScalarRegression(BaseEstimator, RegressorMixin):
+
+    def __init__(self, beta, weights=None):
+        self.beta = beta
+        self.weights = weights
+
+    def fit(self, X, y=None):
+
+        y, X, beta, weights = self._argcheck(y, X)
+
+        nbeta = len(beta)
+        nsamples = X[0].nsamples
+
+        y = np.array(y).reshape((nsamples, 1))
+
+        for j in range(nbeta):
+            xcoef = X[j].coefficients
+            inner_x_beta = X[j].basis.inner_product(beta[j])
+            Zmat = xcoef @ inner_x_beta if j == 0 else np.concatenate(
+                (Zmat, xcoef @ inner_x_beta), axis=1)
+
+        if any(w != 1 for w in weights):
+            rtwt = np.sqrt(weights)
+            Zmat = Zmat * rtwt
+            y = y * rtwt
+
+        Cmat = Zmat.T @ Zmat
+        Dmat = Zmat.T @ y
+
+        Cmatinv = np.linalg.inv(Cmat)
+        betacoefs = Cmatinv @ Dmat
+
+        idx = 0
+        for j in range(0, nbeta):
+            beta[j] = FDataBasis(beta[j], betacoefs[idx:beta[j].nbasis].T)
+            idx = idx + beta[j].nbasis
+
+        self.beta = beta
```

Comment:
That is wrong. The parameters should not be changed, according to Scikit-learn conventions. A `self.beta_` attribute should be assigned to instead (and checked in `predict`).